### PR TITLE
Move to llvmlite 0.32.*

### LINKED ISSUE #5449: bump max llvmlite version to accept 0.32.0
Now that llvmlite 0.32.0rc1 is released. We need to bump the accepted version to `0.33.0.dev0`

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

Closes  #5449

### CODE REVIEW

**Comment by sklam** [comment]
Can we keep allowing 0.31.0? CI has been testing against llvmlite 0.31.0. Supporting 0.31.0 will allow me to release numba wheel first. I have yet to automate the llvmlite wheel building pipeline.

**Comment by stuartarchibald** [comment]
> Can we keep allowing 0.31.0? CI has been testing against llvmlite 0.31.0. Supporting 0.31.0 will allow me to release numba wheel first. I have yet to automate the llvmlite wheel building pipeline.

I expect this will be ok. Specifically, which bits do you think ought to change to accept this, guess `__init__.py` and `setup.py`, and then update the readme?

**Comment by sklam** [comment]
```

**Comment by sklam** [comment]
Hm.. or should it be `<0.33` so that it pick up all the `0.32.*`?

**Comment by sklam** [comment]
ok, here's how we should check:

```python

In [10]: from packaging import version

In [11]: version.parse("0.31") <= version.parse("0.31.0") < version.parse("0.33")
Out[11]: True

In [12]: version.parse("0.31") <= version.parse("0.32.0") < version.parse("0.33")
Out[12]: True

In [13]: version.parse("0.31") <= version.parse("0.32.10") < version.parse("0.33")
Out[13]: True

In [14]: version.parse("0.31") <= version.parse("0.32.0dev0") < version.parse("0.33")
Out[14]: True

In [15]: version.parse("0.31") <= version.parse("0.33.0dev0") < version.parse("0.33")
Out[15]: True

In [16]: version.parse("0.31") <= version.parse("0.33.0rc1") < version.parse("0.33")
Out[16]: True

In [17]: version.parse("0.31") <= version.parse("0.33.0") < version.parse("0.33")
Out[17]: False
```

**Comment by stuartarchibald** [comment]
@sklam are the suggested made on the diff valid with respect to https://github.com/numba/numba/pull/5455#issuecomment-606853691, they seem ok, until we say break an API in `0.33.0dev0`?

File: requirements.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
 setuptools
 numpy>=1.10
-llvmlite==0.31.*
+llvmlite==0.32.*
```
[the remark this answers is not in the document]
Comment (stuartarchibald) [reply requirements.txt:3]: Is `<=0.33` right?

[the remark this answers is not in the document]
Comment (sklam) [reply requirements.txt:3]: changed. it should be `<0.33`

[the remark this answers is not in the document]
Comment (sklam) [reply requirements.txt:3]: that would capture `0.33.0dev<N>` and 0.33.0rc<N>`

File: setup.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 min_numpy_build_version = "1.11"
 min_numpy_run_version = "1.15"
 min_llvmlite_version = "0.31.0.dev0"
-max_llvmlite_version = "0.32.0.dev0"
+max_llvmlite_version = "0.33.0.dev0"
```
Comment (stuartarchibald) [inline setup.py:15]: Wondering if this should be `"0.33.0.rc*"`? Or perhaps move to that later in the release cycle.

Comment (esc) [reply setup.py:15]: My reasoning would be that we don't need to include the RC until we have it, but it's hard to attack this problem with reason, seems like only trial and error will get the job done.

**Comment by esc** [comment]
This made the public CI green again so I am merging this now on the empirical basis that it does seem to fix things.

**Pull request merged**